site/src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,163 @@
+import {
+  ReactElement,
+  ReactNode,
+  cloneElement,
+  createContext,
+  useContext,
+  useRef,
+  useState,
+} from "react";
+// This is used as base for the main Popover component
+// eslint-disable-next-line no-restricted-imports -- Read above
+import MuiPopover, {
+  type PopoverProps as MuiPopoverProps,
+} from "@mui/material/Popover";
+
+type TriggerMode = "hover" | "click";
+
+type TriggerRef = React.RefObject<HTMLElement>;
+
+type TriggerElement = ReactElement<{
+  onClick?: () => void;
+  ref: TriggerRef;
+}>;
+
+type PopoverContextValue = {
+  open: boolean;
+  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
+  triggerRef: TriggerRef;
+  mode: TriggerMode;
+};
+
+const PopoverContext = createContext<PopoverContextValue | undefined>(
+  undefined,
+);
+
+export const Popover = (props: {
+  children: ReactNode | ((popover: PopoverContextValue) => ReactNode); // Allows inline usage
+  mode?: TriggerMode;
+  defaultOpen?: boolean;
+}) => {
+  const [open, setOpen] = useState(props.defaultOpen ?? false);
+  const triggerRef = useRef<HTMLElement>(null);
+  const value = { open, setOpen, triggerRef, mode: props.mode ?? "click" };
+
+  return (
+    <PopoverContext.Provider value={value}>
+      {typeof props.children === "function"
+        ? props.children(value)
+        : props.children}
+    </PopoverContext.Provider>
+  );
+};
+
+export const usePopover = () => {
+  const context = useContext(PopoverContext);
+  if (!context) {
+    throw new Error(
+      "Popover compound components cannot be rendered outside the Popover component",
+    );
+  }
+  return context;
+};
+
+export const PopoverTrigger = (props: {
+  children: TriggerElement;
+  hover?: boolean;
+}) => {
+  const popover = usePopover();
+
+  const clickProps = {
+    onClick: () => {
+      popover.setOpen((open) => !open);
+    },
+  };
+
+  const hoverProps = {
+    onMouseEnter: () => {
```

Comment:
One easy swap that should also help with touch/pointer support is replacing these for `onPointerEnter` and `onPointerLeave`

https://javascript.info/pointer-events#summary

Reply:
Ah, so `onPointEnter` supports touch and mouse events? Is that correct?

Reply:
Yeah, it's a more modern API that tries to roll all three cases into one